client/layout/index.jsx

```diff
@@ -146,7 +146,31 @@ class Layout extends Component {
 		);
 	}
 
+	// This is temporary helper function until we have rolled out to 100% of customers.
+	isNavUnificationEnabled() {
+		if ( ! this.props.teams.length ) {
+			return;
+		}
+
+		// Having the feature enabled by default in all environments, will let anyone use ?flags=-nav-unification to temporary disable it.
+		// We still have the feature disabled in production as safety mechanism for all customers.
+		if ( new URL( document.location ).searchParams.has( 'disable-nav-unification' ) ) {
+			return;
```

Comment:
```suggestion
			return config.disable( 'nav-unification' );
```

Otherwise, the nav unification is enabled after navigating to a different page:

![Jan-26-2021 18-06-12](https://user-images.githubusercontent.com/1233880/105878346-3aa41e00-6001-11eb-981c-22b03e13e39a.gif)

Reply:
Actually, not sure if my suggested change will fix this, since the `isNavUnificationEnabled` function will be re-evaluated in the next render the query param won't be present. Maybe we can fix by moving the `this. isNavUnificationEnabled()` from `render` to `componentDidMount`?

Reply:
Nope nesting it in `componentDidMount` doesn't work consistently cause `teams` is updated after the layout mounts. What I did is:
- use `UNSAFE_componentWillMount` for initial load (eg `teams` is already in indexed DB)
- use `componentDidUpdate` for catching `teams` update and only call `this.isNavUnificationEnabled();` if the `teams` object has changed.

7c46153